packages/tasks-gen/scripts/generate-snippets-fixtures.ts

```diff
@@ -0,0 +1,178 @@
+/*
+ * Generates Inference API snippets using @huggingface/tasks snippets.
+ *
+ * If used in test mode ("pnpm test"), it compares the generated snippets with the expected ones.
+ * If used in generation mode ("pnpm generate-snippets-fixtures"), it generates the expected snippets.
+ *
+ * Expected snippets are saved under ./snippets-fixtures and are meant to be versioned on GitHub.
+ * Each snippet is saved in a separate file placed under "./{test-name}/{index}.{client}.{language}":
+ *   - test-name: the name of the test (e.g. "text-to-image", "conversational-llm", etc.)
+ *   - index: the order of the snippet in the array of snippets (0 if not an array)
+ *   - client: the client name (e.g. "requests", "huggingface_hub", "openai", etc.). Default to "default" if client is not specified.
+ *   - language: the language of the snippet (e.g. "sh", "js", "py", etc.)
+ *
+ * Example:
+ *   ./packages/tasks-gen/snippets-fixtures/text-to-image/0.huggingface_hub.py
+ */
+
+import { existsSync as pathExists } from "node:fs";
+import * as fs from "node:fs/promises";
+import * as path from "node:path/posix";
+
+import type { InferenceSnippet } from "@huggingface/tasks";
+import { snippets } from "@huggingface/tasks";
+
+type LANGUAGE = "sh" | "js" | "py";
```

Comment:
for the info, this is how the same thing is declared in moon. In subseq PR, we can probably declare `InferenceSnippetLanguage` in `tasks`and use it in `tasks-gen` & `moon` 

```ts
export const inferenceSnippetLanguages = ["python", "js", "curl"] as const;

export type InferenceSnippetLanguage = (typeof inferenceSnippetLanguages)[number];
```